app/services/katello/content_unit_indexer.rb

```diff
@@ -195,6 +188,27 @@ def clean_filter_rules(repo_associations_to_destroy)
       end
     end
 
+    def clean_duplicate_docker_tags
+      # Deduplicate docker tags by name, keeping the one with the newest updated_at
+      deduplicated_tags = @repository.docker_tags.group_by(&:name).map { |_name, tags| tags.max_by(&:updated_at) }
+
+      deduplicated_tags.each do |valid_tag|
+        # Remove associations for each manifest containing a tag with the same name as `valid_tag`, except the newest
+        tag_ids_to_remove = @repository.docker_manifests
+                          .joins(:docker_tags)
+                          .where(katello_docker_tags: {name: valid_tag.name})
+                          .where.not(id: valid_tag.docker_taggable_id)
+                          .flat_map(&:docker_tags)
+                          .select { |tag| tag.name == valid_tag.name }
+                          .map(&:id)
+
```

Comment:
This seems like it would take all unique tags in repo and then query tags again in the loop to find dups..There could be a way to get the duplicate ids from sql and not have these additional queries per tag. Also the deletion is inside the inner loop..We could move it out to make one delete call outside the loop.